### PR TITLE
[FLINK-6222] Allow passing env variables to start scripts via

### PR DESCRIPTION
**Pull request by dawidwys**
Added possibility to pass env variables (e.g. HADOOP_CONF_DIR) through flink-env.sh

### CODE REVIEW

**Comment by dawidwys** [comment]
I added a small section. I was thorn a little in the first place if we should make it highly visible in docs as it might be wrongly understood those variables will be distributed to the cluster as well, which is not true. What do you think?

**Comment by dawidwys** [comment]
In the first place I thought it might make providing the variables a bit more flexible with `flink-env.sh`. In the end it is just a bash script, but the longer I think the more convinced I am it was not the right choice. As you said it does not go well with the other opts, plus we might confuse some users. 

I will close this PR and open new one with `env.hadoop.home` tomorrow.

File: flink-dist/src/main/flink-bin/bin/config.sh [hunk 1]
```diff
@@ -96,6 +96,8 @@ DEFAULT_ENV_JAVA_OPTS=""                            # Optional JVM args
 DEFAULT_ENV_JAVA_OPTS_JM=""                         # Optional JVM args (JobManager)
 DEFAULT_ENV_JAVA_OPTS_TM=""                         # Optional JVM args (TaskManager)
 DEFAULT_ENV_SSH_OPTS=""                             # Optional SSH parameters running in cluster mode
+DEFAULT_YARN_CONF_DIR=""            # YARN Configuration Directory, if necessary
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/config.sh]: indentation?

**Comment by dawidwys** [comment]
Thanks for the review @StefanRRichter and @zentol. Will fix the indentation while merging.

**Pull request closed without merging**